packages/core/helper-plugin/lib/src/components/DynamicTable/tests/Table.test.js

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable react/prop-types */
+import {
+  BaseCheckbox,
+  Tbody,
+  Td,
+  ThemeProvider,
+  Tr,
+  Typography,
+  lightTheme,
+} from '@strapi/design-system';
+import { render, fireEvent, waitFor, screen } from '@testing-library/react';
+import * as React from 'react';
+import { IntlProvider } from 'react-intl';
+import { MemoryRouter } from 'react-router-dom';
+import TrackingProvider from '../../../providers/TrackingProvider';
+import Table from '../index';
+
+const TableRows = ({
+  entriesToDelete,
+  onSelectRow,
+  rows,
+  headers,
+  withMainAction,
+  allowRangeSelection = true,
+}) => {
+  return (
+    <Tbody>
+      {rows.map((data, index) => {
+        const isChecked = entriesToDelete.findIndex((id) => id === data.id) !== -1;
+
+        const handleRowSelection = (event) => {
+          onSelectRow({
+            name: data.id,
+            value: !isChecked,
+            index: allowRangeSelection ? index : undefined,
+            isShiftKeyHeld: allowRangeSelection ? event.nativeEvent.shiftKey : undefined,
+          });
+        };
+
+        return (
+          <Tr key={data.id}>
+            {withMainAction && (
+              <Td>
+                <BaseCheckbox
+                  aria-label={`Select ${data.firstname} ${data.lastname}`}
+                  className={isChecked ? 'selected' : ''}
+                  checked={isChecked}
+                  onChange={handleRowSelection}
+                />
+              </Td>
+            )}
+
+            {headers.map(({ key, name }) => {
+              return (
+                <Td key={key}>
+                  <Typography textColor="neutral800">{data[name] || '-'}</Typography>
+                </Td>
+              );
+            })}
+          </Tr>
+        );
+      })}
+    </Tbody>
+  );
+};
+
+const renderTable = (headers, rows, allowRangeSelection = true, withMainAction = true) =>
+  render(
+    <MemoryRouter>
+      <TrackingProvider value={{ uuid: '2479d6d7-2497-478d-8a34-a9e8ce45f8a7' }}>
+        <ThemeProvider theme={lightTheme}>
+          <IntlProvider locale="en" messages={{}} textComponent="span">
+            <Table
+              headers={headers}
+              contentType="users"
+              rows={rows}
+              withMainAction={withMainAction}
+            >
+              <TableRows headers={headers} rows={rows} allowRangeSelection={allowRangeSelection} />
+            </Table>
+          </IntlProvider>
+        </ThemeProvider>
+      </TrackingProvider>
+    </MemoryRouter>
+  );
```

Comment:
you should only render `Table/TableRows` the rest should use the `wrapper` option – https://testing-library.com/docs/react-testing-library/api/#wrapper